go/vt/vttest/environment.go

```diff
@@ -231,9 +233,19 @@ func tmpdir(dataroot string) (dir string, err error) {
 	return
 }
 
+// randomPort gets a random port that is available for a TCP connection.
+// After we generate a random port, we try to establish a tcp connection on it.
+// If it fails, then we try a different port.
 func randomPort() int {
-	v := rand.Int32N(20000)
-	return int(v + 10000)
+	for {
+		port := int(rand.Int32N(20000) + 10000)
+		ln, err := net.Listen("tcp", net.JoinHostPort("127.0.0.1", strconv.Itoa(port)))
+		if err != nil {
+			continue
+		}
+		ln.Close()
```

Comment:
I think this could still race as it can return the same available port if we run this in parallel but I don't know if that's something that happens in practice?

Reply:
Also, I guess there can still be race conditions on the other ports? If I understand this correctly, this is just the base port, and then other services just increment that port number - so when this test is run, we don't know whether one of the other ports won't clash with something else either.

Reply:
Yes, good point! Fixed it now.